server/index.js

```diff
@@ -1,24 +1,49 @@
 const express = require('express');
 const {dbConnection} = require('../MYSQL/index.js');
 const morgan = require('morgan');
-const path = require('path');
+const {join} = require('path');
 const cors = require('cors');
 
 const app = express();
 app.use(cors());
-app.use(express.static('public/dist'));
+// app.use(express.static('public/dist'));
 
 app.use(express.json());
 app.use(express.urlencoded({extended: true}));
-app.use(morgan('combined'))
+app.use(morgan('combined'));
+
 
 app.use((req,res,next) =>{
   res.header('Access-Control-Allow-Origin', '*');
   res.header('Access-Control-Headers','Origin','X-Requested-With','Content-Type','Accept')
   next();
 })
+
 app.use('/:listingId/', express.static('public/dist'));
 
+
+
+
+app.get('/bundle.js',(req,res) =>{
+
+  if(req.header('Accept-Encoding').includes('br')) {
+    console.log('calling brotli');
+    res.set('Content-Encoding', 'br');
+    res.set('Content-Type','application/javascript');
+    res.sendFile(join(__dirname,'../','public','dist','bundle.js.br'))
+  } else if(req.header('Accept-Encoding').incldues('gz')){
+    console.log('calling gzip');
+    res.set('Content-Encoding','gzip');
+    res.set('Content-Type','application/javascript');
+    res.sendFile(join(__dirname,'../','public','dist','bundle.js.gz'))
+  }else{
+    console.log('calling uncompressed');
+    res.sendFile(join(__dirname,'../','public','dist','bundle.js'))
+  }
+});
```

Comment:
Whats happening here? So you're sending a different compressed file depending on what the browser expectS?

Reply:
From my understanding not all browsers accept br encoding. Chrome and other known browsers should accept br but i just wanted to make it work on all browsers

Reply:
By how much did this improve your page speed?